cfme/middleware/server.py

```diff
@@ -46,12 +62,44 @@ def __init__(self, name, provider=None, **kwargs):
 
     @classmethod
     def servers(cls, provider):
-        sel.force_navigate('middleware_servers')
-        deployments = []
+        servers = []
+        if provider:
+            # if provider instance is provided try to navigate provider's servers page
+            # if no servers are registered in provider, returns empty list
+            if provider.load_all_provider_servers():
```

Comment:
I would expect that if `load_all_provider_servers()` returns True then there are some servers but in this case it returns an empty list.